conn.go

```diff
@@ -25,25 +24,24 @@ func beginHandlerOpen(ch chan error, removeHandlers func()) func(ev *js.Object)
 
 // closeError allows a CloseEvent to be used as an error.
 type closeError struct {
-	*dom.CloseEvent
+	*js.Object
```

Comment:
I think making closeError embed the correct JS fields makes the code much more readable and concise (since it removes the need to have ```xx.Get(yy).Type()``` calls everywhere. However I am unsure about the messageEvent change. It actually makes it more clear what is passed around, but makes the code a little more verbose. But I think this verbosity is worth the clarity brings around. I separated both changes in two commits, so you can review them separately. :smile: